apm-agent-plugins/apm-servlet-plugin/src/main/java/co/elastic/apm/agent/servlet/ServletInitInstrumentation.java

```diff
@@ -0,0 +1,113 @@
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.NamedElement;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nullable;
+import javax.servlet.ServletConfig;
+import javax.servlet.ServletContext;
+import java.util.Collection;
+import java.util.Collections;
+
+import static co.elastic.apm.agent.servlet.ServletInstrumentation.SERVLET_API;
+import static net.bytebuddy.matcher.ElementMatchers.hasSuperType;
+import static net.bytebuddy.matcher.ElementMatchers.isInterface;
+import static net.bytebuddy.matcher.ElementMatchers.nameContains;
+import static net.bytebuddy.matcher.ElementMatchers.nameContainsIgnoreCase;
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static net.bytebuddy.matcher.ElementMatchers.not;
+import static net.bytebuddy.matcher.ElementMatchers.takesArgument;
+
+/**
+ * Instruments {@link javax.servlet.Servlet#init(ServletConfig)} to provide a warning when an unsupported Servlet version is used
+ */
+public class ServletInitInstrumentation extends ElasticApmInstrumentation {
+
+    @VisibleForAdvice
+    public static final Logger logger = LoggerFactory.getLogger(ServletTransactionHelper.class);
+
+    /**
+     * Allows to perform check only once with loose concurrency requirements. Thus there might be multiple warning
+     * messages, but most of the time there won't.
+     */
+    @VisibleForAdvice
+    public static volatile boolean doCheckAndWarn = true;
+
+    @Override
+    public ElementMatcher<? super NamedElement> getTypeMatcherPreFilter() {
+        return nameContains("Servlet").or(nameContainsIgnoreCase("jsp"));
+    }
+
+    @Override
+    public ElementMatcher<? super TypeDescription> getTypeMatcher() {
+        return not(isInterface())
+            .and(hasSuperType(named("javax.servlet.Servlet")));
+    }
+
+    @Override
+    public ElementMatcher<? super MethodDescription> getMethodMatcher() {
+        return named("init")
+            .and(takesArgument(0, named("javax.servlet.ServletConfig")));
+    }
+
+    @Override
+    public Collection<String> getInstrumentationGroupNames() {
+        return Collections.singleton(SERVLET_API);
+    }
+
+    @Advice.OnMethodEnter(suppress = Throwable.class)
+    private static void onEnter(@Advice.Argument(0) @Nullable ServletConfig servletConfig) {
+
+        if (!doCheckAndWarn) {
+            return;
+        }
+
+        int majorVersion = -1;
+        int minorVersion = -1;
+        if (servletConfig != null) {
+            ServletContext servletContext = servletConfig.getServletContext();
+            if (null != servletContext) {
+                majorVersion = servletContext.getMajorVersion();
+                minorVersion = servletContext.getMinorVersion();
+            }
+        }
+
```

Comment:
Kill two birds with one stone and also log the servlet info. That will be really useful to avoid back-and-forth in support requests.

```java
logger.info(servletConfig.getServletContext().getServerInfo());
```